src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/hooks.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+import { useFlatFormErrors } from '~hooks';
+import { ACTION, Action } from '~constants/actions';
+
+export const useGetActionErrors = () => {
+  const { formState } = useFormContext();
+  const hasErrors = !formState.isValid && formState.isSubmitted;
+  const flatFormErrors = useFlatFormErrors(formState.errors).filter(
+    ({ key }) => key !== 'title',
+  );
+
+  return {
+    hasErrors,
+    flatFormErrors,
+  };
+};
+
+export const useGetAdditionalValidationErrors = (
+  action: Action | undefined,
+) => {
+  const {
+    formState: { isDirty },
+  } = useFormContext();
+
+  if (!action) {
+    return undefined;
+  }
+
+  switch (action) {
+    case ACTION.EDIT_COLONY_DETAILS:
+      return isDirty ? undefined : (
+        <FormattedMessage id="actionSidebar.editColonyDetails.error" />
```

Comment:
could we maybe just return the string here? I know `FormattedMessage` basically generates a string, but it's not common practice to return JSX from hooks